File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
    Map<Integer, Integer> map = new HashMap<>();
    map.put(0,1);
    int oddCount = 0;
    int ans = 0;
    int n = nums.length;
    for(int i = 0;i<n;i++){
      if(nums[i]%2 == 1){
        oddCount++;
      }
      map.put(oddCount,map.getOrDefault(oddCount,0)+1);
      ans = ans + map.getOrDefault(oddCount-k, 0);
        
    }
        return ans;  
    }
}

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& A, int k) {
        int n = A.size();
  vector<int>oddInd;
  oddInd.push_back(-1);
  for(int i=0; i<n; i++){
    if(A[i]%2){
      oddInd.push_back(i);
    }
  }
  oddInd.push_back(n);
  int m = oddInd.size();
  if(m-2 < k){
    return 0;
  }
  int ans = 0;
  int j = 1;
  for(int i=k+1; i<m; i++){
    ans += (oddInd[i] - oddInd[i-1])*(oddInd[j] - oddInd[j-1]);
    j++;
  }
  return ans;
    }
};